web/frontend/src/components/slam-2d-render.vue

```diff
@@ -63,25 +72,96 @@ controls.enableRotate = false;
 
 const raycaster = new MouseRaycaster({ camera, renderer, recursive: false });
 
-raycaster.on('click', (event) => {
+raycaster.on('click', (event: THREE.Event) => {
   const [intersection] = event.intersections as THREE.Intersection[];
   if (intersection && intersection.point) {
-    console.log(intersection.point);
+    emit('click', intersection.point);
   }
 });
 
-const markerSize = 0.5;
-const marker = new THREE.Mesh(
-  new THREE.PlaneGeometry(markerSize, markerSize).rotateX(-Math.PI / 2),
-  new THREE.MeshBasicMaterial({ color: 'red' })
-);
-marker.name = 'Marker';
-// This ensures the robot marker renders on top of the pointcloud data
-marker.renderOrder = 999;
+const guiData = {
+    drawFillShapes: true,
+    drawStrokes: true,
+    fillShapesWireframe: false,
+    strokesWireframe: false,
+  };
+/*
+ * svgLoader example for webgl:
+ * https://github.com/mrdoob/three.js/blob/master/examples/webgl_loader_svg.html
+ */
+//
+const svgLoader = new SVGLoader(); 
+const makeMarker = async (url : string, name: string, scalar: number) => {
+  const data = await svgLoader.loadAsync(url);
+
+  const { paths } = data!;
+
+  const group = new THREE.Group();
+  group.scale.multiplyScalar(scalar);
+  group.position.set(-70, 0, 70) // why do we do this? - do we need it?
+  group.scale.y *= -1; // why do we do this? - do we need it?
+
+  for (const path of paths) {
+
+    const fillColor = path!.userData!.style.fill;
+
+    if (guiData.drawFillShapes && fillColor !== undefined && fillColor !== 'none') {
+      const material = new THREE.MeshBasicMaterial({
+        color: new THREE.Color().setStyle(fillColor)
+          .convertSRGBToLinear(),
+        opacity: path!.userData!.style.fillOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
+        depthWrite: false,
+        wireframe: guiData.fillShapesWireframe,
+      });
+
+      const shapes = SVGLoader.createShapes(path!);
+
+      for (const shape of shapes) {
+
+        const geometry = new THREE.ShapeGeometry(shape);
+        const mesh = new THREE.Mesh(geometry.rotateX(-Math.PI / 2).rotateY(Math.PI), material);
+        group.add(mesh);
+
+      }
+
+    }
+
+    const strokeColor = path!.userData!.style.stroke;
+
+    if (guiData.drawStrokes && strokeColor !== undefined && strokeColor !== 'none') {
+
+      const material = new THREE.MeshBasicMaterial({
+        color: new THREE.Color().setStyle(strokeColor)
+          .convertSRGBToLinear(),
+        opacity: path!.userData!.style.strokeOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
+        depthWrite: false,
+        wireframe: guiData.strokesWireframe,
+      });
+
+      for (let j = 0, jl = path!.subPaths.length; j < jl; j += 1) {
```

Comment:
i am unsure how this would become a `for .. of` loop

Reply:
```ts
for (const subPath of path!.subPaths) {
  ...
}
```